src/Sentry/BaggageHeader.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Sentry.Internal.Extensions;
+
+namespace Sentry
+{
+    /// <summary>
+    /// Baggage Header for dynamic sampling.
+    /// </summary>
+    /// <seealso href="https://develop.sentry.dev/sdk/performance/dynamic-sampling-context"/>
+    /// <seealso href="https://www.w3.org/TR/baggage"/>
+    internal class BaggageHeader
+    {
+        internal const string HttpHeaderName = "baggage";
+        internal const string SentryKeyPrefix = "sentry-";
+
+        // https://www.w3.org/TR/baggage/#baggage-string
+        // "Uniqueness of keys between multiple list-members in a baggage-string is not guaranteed."
+        // "The order of duplicate entries SHOULD be preserved when mutating the list."
+
+        public IReadOnlyList<KeyValuePair<string, string>> Members { get; }
+
+        private BaggageHeader(IEnumerable<KeyValuePair<string, string>> members) =>
+            Members = members.ToList().AsReadOnly();
+
+        // We can safely return a dictionary of Sentry members, as we are in control over the keys added.
+        // Just to be safe though, we'll group by key and only take the first of each one.
+        public IReadOnlyDictionary<string, string> GetSentryMembers() =>
+            Members
+                .Where(kvp => kvp.Key.StartsWith(SentryKeyPrefix))
+                .GroupBy(kvp => kvp.Key, kvp => kvp.Value)
+                .ToDictionary(
+#if NETCOREAPP || NETSTANDARD2_1
+                    g => g.Key[SentryKeyPrefix.Length..],
+#else
+                    g => g.Key.Substring(SentryKeyPrefix.Length),
+#endif
+                    g => g.First());
+
+        /// <summary>
+        /// Creates the baggage header string based on the members of this instance.
+        /// </summary>
+        /// <returns>The baggage header string.</returns>
+        public override string ToString()
```

Comment:
not a fain of overriding string. would prefer to make it explicit as to the use case. eg `ToHeaderString`

Reply:
I hear ya, but I think I'll keep it on `ToString`, as it helps when debugging and also we already are doing that for `SentryTraceHeader` and a few other similar objects.  Thanks.